Lib/test/test_unicode.py

```diff
@@ -2807,6 +2807,25 @@ def check_format(expected, format, *args):
         check_format('repr=abc',
                      b'repr=%V', 'abc', b'xyz')
 
+        # test %p
+        # We cannot test the exact result,
+        # because it returns a hex representation of a C pointer,
+        # which is going to be different each time. But, we can test the format.
+        p_format_regex = r'0x[a-zA-Z0-9]{8,}'
```

Comment:
```suggestion
        p_format_regex = r'^0x[a-zA-Z0-9]{8,}$'
```
Make sure it matches the whole string